sdk/storage/azure-storage-blob/azure/storage/blob/_encryption.py

```diff
@@ -40,6 +40,9 @@
 
 _ENCRYPTION_PROTOCOL_V1 = '1.0'
 _ENCRYPTION_PROTOCOL_V2 = '2.0'
+_ENCRYPTION_PROTOCOL_V2_1 = '2.1'
+_VALID_ENCRYPTION_PROTOCOLS = [_ENCRYPTION_PROTOCOL_V1, _ENCRYPTION_PROTOCOL_V2, _ENCRYPTION_PROTOCOL_V2_1]
```

Comment:
We check 
`if version not in (_ENCRYPTION_PROTOCOL_V1, _ENCRYPTION_PROTOCOL_V2, _ENCRYPTION_PROTOCOL_V2_1)`
in two different places, and in my initial sweep I only caught one and not the other. I think this would be a good idea to unify them in a list, so we can simply modify it in one place and not have to track all instances

Reply:
nit: The other check was actually:
L553: `if protocol not in [_ENCRYPTION_PROTOCOL_V1, _ENCRYPTION_PROTOCOL_V2]:`

Super nit but it would have probably been easier to catch if we consistently used `protocol` or `version`. I see that we only really have one instance of a `protocol` local variable (L555) and everywhere else we take it as input as `version`. I wouldn't be opposed to completely removing use of "protocol" and switching it to "version" to avoid confusion that it's something fundamentally different, and would also make (albeit hopefully rare) refactors less error-prone 😄